casper/src/main/scala/coop/rchain/casper/BlockRandomSeed.scala

```diff
@@ -38,4 +57,72 @@ object BlockRandomSeed {
 
   def generateRandomNumber(blockRandomSeed: BlockRandomSeed): Blake2b512Random =
     Blake2b512Random(encode(blockRandomSeed))
+
+  def fromBlock(block: BlockMessage): Blake2b512Random = {
+    val seed = BlockRandomSeed(
+      block.shardId,
+      block.blockNumber,
+      PublicKey(block.sender),
+      block.preStateHash.toBlake2b256Hash
+    )
+    generateRandomNumber(seed)
+  }
+
+  def fromGenesis(block: BlockMessage): Blake2b512Random =
+    fromGenesis(block.shardId)
+
+  def fromGenesis(shardId: String): Blake2b512Random = {
+    val seed = BlockRandomSeed(
+      shardId,
+      GenesisRandomSeedBlockNumber,
+      GenesisRandomSeedPubKey,
+      EmptyBytesBlakeHash
```

Comment:
`GenesisRandomSeedPubKey`, `GenesisRandomSeedBlockNumber` are used only here for this purpose so they don't even deserve to have their own name. :)

Reply:
What name do you perfer?

Reply:
I surrender completely to your paranoia on names.

Reply:
No name, just add the values here. These names are not used anywhere else except in this method.